magento/graphql-ce
# Add endpoint for fetching/deleting vault tokens

https://github.com/magento/graphql-ce/issues/46

### Description (*)
Add query for fetching visible payment tokens:
```graphql
customerPaymentTokens {
     items {
        public_hash
        details
        payment_method_code
        type
    }
}
```
The `details` property is the string value of `vault_payment_token.details`. This field contains serialized details that vary by the payment implementation. I'm not sure if we want to expose an interface for formatting this information similar to what is used in Magento_InstantPurchase or leave as-is.

Add mutation for deleting payment tokens:
```graphql
deletePaymentToken (
    public_hash: {{someHash}}
) {
    result
}
```

### Fixed Issues (if relevant)
1. magento/graphql-ce#46: [Query] My Account > Stored Payment Methods

### Manual testing scenarios (*)
1. Configure payment method supporting vault and enable
2. Checkout via storefront or admin using payment method and select option to save for later
3. Use query above to retrieve stored payment tokens via graphql **requires authentication**
4. Use mutation above to delete stored payment token via graphql **requires authentication**
5. Use query above to validate deleted payment token is no longer returned via graphql **requires authentication**

### Contribution checklist (*)
